opentelemetry_api/opentelemetry/context/traceparent.c

```diff
@@ -0,0 +1,106 @@
+/* Copyright 2019, OpenTelemetry Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#include "traceparent.h"
+
+status_t _uint_from_hex_string(unsigned int* value, const char* s) {
```

Comment:
Technically "status_t" not part of POSIX. Anything that has **_t** in type name is reserved for POSIX.
Cit.: https://pubs.opengroup.org/onlinepubs/9699919799/functions/V2_chap02.html
See section about the "Suffix" -- _t reserved for ANY HEADER..

I think many good open source libraries don't necessarily follow that rule, e.g.
- nginx widely uses _t suffixes.
- Android Framework uses _status_t_ suffix for a non-POSIX type.. In fact ours would then clash with Android's definition!
- Windows SDK uses _t suffix for C-types in several SDK headers..

But the best practice would probably either avoid using __t_ suffix, or ensure that all C-API types have a unique prefix, such as _otel__ or at least _ot_

Reply:
I vote for not using *_t suffix on our datatypes. I'd prefer `opentelemetry::Status`.